kernel/src/syscall/mmap.rs

```diff
@@ -57,6 +57,9 @@ fn do_sys_mmap(
     if len == 0 {
         return_errno_with_message!(Errno::EINVAL, "mmap len cannot be zero");
     }
+    if len > usize::MAX - PAGE_SIZE + 1 {
```

Comment:
First, are you sure that `len == usize::MAX - PAGE_SIZE + 1` is ok? This case the subsequent `align_up` will also overflow.

Second, I think we can choose a much lower upper bound for `len`. For example, `isize::MAX as usize`. This value is good for two reasons: (1) On x86-64, the lower half of the whole address space is reserved for the user space. So it is impossible for a user program to operate on a memory region whose length is beyond `isize::MAX as usize`. (2) Sometimes, it would be convenient to convert `usize` to `isize`. Using ``isize::MAX` as the upper bound ensures that it is ok to convert such a `usize` value to `isize`.

Reply:
1. If I do not understand wrong, the `align_up` for `len: usize` is `len.checked_add(align - 1).unwrap()`, whose overflow condition is  `len + PAGE_SIZE - 1 > usize::MAX`. So `len` won't overflow when `len <= usize::MAX - PAGE_SIZE + 1` 
https://github.com/asterinas/asterinas/blob/a72c7dadf3b9bb680b3092af6b64084b98008cb7/ostd/libs/align_ext/src/lib.rs#L61

Reply:
2. Yes, in linux they use a const value `TASK_SIZE` to limit the range, which is `0xFFFFFFFFUL`.
https://elixir.bootlin.com/linux/v6.10.5/source/mm/mmap.c#L1864

Reply:
Using `isize::MAX  / 2` looks fine with me.

Reply:
If the boundary is set to `isize::MAX  / 2`, the CI will fail
https://github.com/asterinas/asterinas/actions/runs/10612897770/job/29418119617
And the result could be reproduced locally.

So I keep the boundary to `isize::MAX`

Reply:
> If the boundary is set to `isize::MAX / 2`, the CI will fail https://github.com/asterinas/asterinas/actions/runs/10612897770/job/29418119617 And the result could be reproduced locally.
> 
> So I keep the boundary to `isize::MAX`

Sorry. I am not thinking straight. I meant `isize::MAX`.